core/io/file_access_pack.cpp

```diff
@@ -447,7 +447,11 @@ void FileAccessPack::set_big_endian(bool p_big_endian) {
 }
 
 Error FileAccessPack::get_error() const {
-	if (eof) {
+	if (encryption_error) {
+		return ERR_UNAUTHORIZED;
```

Comment:
I'm not sure if unauthorized is the right message to send here. I'd probably return `ERR_FILE_CORRUPT` or `ERR_FILE_CANT_READ`.